src/libraries/Microsoft.NETCore.Platforms/src/AssemblyResolver.cs

```diff
@@ -38,7 +38,10 @@ private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEven
             Assembly assm = null;
 
             // look next to requesting assembly
+            // The code has a fallback to use AppDomain.CurrentDomain.BaseDirectory so is not single file dangerous
+#pragma warning disable IL3000 // Avoid accessing Assembly file path when publishing as a single file
```

Comment:
Would it be possible to use the `UnconditionalSuppressMessage` attribute to do these suppressions?

From my past experience, every time someone hit an issue with APIs like `Assembly.Location` and couldn't figure it out, it was in a third party NuGet. If they hit it in their own code, they could usually just debug into it. E.g. the CoreRT repo is full of them; here's one: https://github.com/dotnet/corert/issues/7756.

I think there's a lot of value in eventually detecting this in compiled assemblies as well (either with illink or some other tool). We're not doing that right now. Those tools will not see the `#pragama warning` suppressions, so we'll have to redo it.

E.g. I could add this detection to the NativeAOT compiler now, but if the suppressions in the framework are only for the analyzer, there's little point to do that.

Reply:
Changed to UnconditionalSuppressedMessage since a new version of arcade that include the roslyn changes was merged to runtime